### PR TITLE
GH actions nightly builds

### PR DESCRIPTION
**Pull request by Mab879**
#### Description:

Adds a nightly build script for the content.

#### Rationale:

Move things off of Jenkins

### CODE REVIEW

**Comment by Mab879** [comment]
[Example build](https://github.com/Mab879/content/actions/runs/1780559755)

**Comment by Mab879** [comment]
So there is few issues with using GitHub actions for nightly builds:
1. No easy way to directly link to the latest build
2. A user must be signed in order to get the build

**Comment by evgenyz** [comment]
> So there is few issues with using GitHub actions for nightly builds:
>     1. No easy way to directly link to the latest build
>     2. A user must be signed in order to get the build

Can https://nightly.link/ help? I know that it's a workaround, but we are on GH not GL, changes don't happen overnight here...

**Comment by jan-cerny** [comment]
It seems that it would address the problem.

**Comment by Mab879** [comment]
#8734 needs to be fixed for there to be a 5.10 build.

**Comment by Mab879** [comment]
Once this is merged, I will update the README with https://nightly.link/

File: .github/workflows/nightly_build.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: Nightly builds
+on:
+    schedule:
+        # Run daily at 03:00
+        -   cron: "0 3 * * *"
+    push:
+        branches:
+            - gh_actions_nightly_builds
```
Comment (ggbecker) [inline .github/workflows/nightly_build.yml:8]: are `branches` here really needed?

**Review by ggbecker — APPROVED** [review]

**Comment by yuumasato** [comment]
@Mab879 @ggbecker It seems we have some issue with the syntax:
https://github.com/ComplianceAsCode/content/actions/workflows/nightly_build_5_10.yml

**Comment by Mab879** [comment]
https://github.com/mab879/content/actions/workflows/nightly_build_5_10.yml is the latest build from my repo. We may still have validation issues for the OVAL 5.10 content.

**Pull request merged**